lib/util.js

```diff
@@ -221,26 +223,32 @@ function format(f) {
           str += inspect(arguments[a++],
                          { showHidden: true, depth: 4, showProxy: true });
           break;
-        case 37: // '%'
-          str += '%';
+        case 105: // 'i'
+          str += parseInt(arguments[a++]);
           break;
+        case 102: // 'f'
+          str += parseFloat(arguments[a++]);
+          break;
+        case 37: // '%'
+          i++;
+          lastPos = i;
+          continue;
         default: // any other character is not a correct placeholder
-          str += '%';
-          lastPos = i = i + 1;
+          lastPos = i;
+          i++;
           continue;
       }
-      lastPos = i = i + 2;
-      continue;
+      i++;
+      lastPos = i + 1;
     }
-    ++i;
   }
   if (lastPos === 0)
     str = f;
   else if (lastPos < f.length)
     str += f.slice(lastPos);
   while (a < arguments.length) {
     const x = arguments[a++];
-    if (x === null || (typeof x !== 'object' && typeof x !== 'symbol')) {
+    if (typeof x !== 'object' && typeof x !== 'symbol' || x === null) {
```

Comment:
I think this is harder to reason about, which is why I always add extra parentheses so it's always immediately clear (especially to newcomers).

Reply:
It is interesting to see many people preferring this style. I personally always feel like my brain has to compute a tiny bit more when there are extra braces^^. I changed it back though.